10539 export.rst doc edit

Konntroll opened the pull request:
Finished editing export.rst for task #10539.

Please let me know if you would like me to make any changes to the text.

tcmetzger commented on sphinx/source/docs/user_guide/export.rst:
```diff
@@ -1,46 +1,44 @@
 
-In order to use the |export| functions, users may have to install additional
-dependencies. One option is to install them via conda:
+You may need additional dependencies to use the |export| functions.
```
```suggestion
You may need to install additional dependencies to use the |export| functions.
```

tcmetzger commented on sphinx/source/docs/user_guide/export.rst:
```diff
@@ -68,35 +65,33 @@ saving the image to a file, you can use the lower-level function
-Bokeh also supports replacing the HTML5 Canvas plot output with an SVG element
-that can be edited in image editing programs such as Adobe Illustrator and/or
-converted to PDFs.
+Bokeh can also replace HTML5 Canvas plot output with an Scalable Vector
```
```suggestion
Bokeh can also replace the HTML5 Canvas plot output with a Scalable Vector
```

tcmetzger commented on sphinx/source/docs/user_guide/export.rst:
```diff
@@ -68,35 +65,33 @@ saving the image to a file, you can use the lower-level function
-in the browser.
+You can't create a single SVG for a layout of plots because each plot will
+produce its own distinct SVG element. You can, however, download an SVG plot
+using a SaveTool from the toolbar although an SVG export of a plot with a
```
very long sentence with lots of modifiers. Maye somthing like: "... using a SaveTool from the toolbar. If you use this option to create an SVG export and your plot contains a toolbar, keep in mind that the exported file will have a blank area ..."

Konntroll replied:
Okay, yeah. This one is a bit heavy-handed. Broke it up and simplified.

bryevdv commented:
@Konntroll similar comments to the other PR, these changes have file deletions which will need to be reverted. You can try to use `git revert` as described on the other PR, or submit a new clean PR. Just make sure you check out `branch-2.3` *first*, and *then* create your new PR branch, so that the changes in the PR are relative to `branch-2.3`

bryevdv commented on sphinx/source/docs/user_guide/export.rst:
```diff
@@ -68,35 +65,33 @@ saving the image to a file, you can use the lower-level function
-should set the ``Plot.background_fill_color`` and ``Plot.border_fill_color``
-properties to ``None``.
+To create an SVG with a transparent background, set the ``Plot.background_fill_color``
+and ``Plot.border_fill_color`` properties to ``None``, same as with PNG output.
```
```suggestion
and ``Plot.border_fill_color`` properties to ``None``.
```
I'm not sure the callback reference to PNG adds anything.

bryevdv commented on sphinx/source/docs/user_guide/export.rst:
```diff
@@ -68,35 +65,33 @@ saving the image to a file, you can use the lower-level function
+You can't create a single SVG for a layout of plots because each plot will
+produce its own distinct SVG element. You can, however, download an SVG plot
+using a SaveTool from the toolbar. Note that in this case the exported file
+will have a blank area where the toolbar used to be.
```
I think this entire para needs more attention. These two cases don't really have anything to do with one another, but it reads (since the original) like somehow the savetool is a solution to the first issue with layouts. 

Suggest that the last two sentences be combined:

> You can download an SVG plot using a SaveTool from the toolbar (note that the exported file
will have a blank area where the toolbar was).

And I think this should be separated from the first sentence as well, either two very short one-sentence para's or turn in to a bulleted list maybe?

@tcmetzger about this:

> You can't create a single SVG for a layout of plots because each plot will produce its own distinct SVG element.

I wanted to see if you have any suggestion for different wording. This is a limitation of the library, but in this context the "You can't ..." reads to me like "You aren't allowed" TBH I prefer direct statements like "It is not possible to..." in cases where things are.... not possible. What is the compromise update here?

bryevdv replied:
Oh I see, that original "Alternatively,..." looks to be copy-pasta from another section. It was definitely badly worded from the start.

tcmetzger replied:
Well, "it is not possible" sounds pretty good to me! Or you could turn around the sentence a bit: "Each plot is its own SVG element. Therefore, it is not possible to automatically create a single SVG combining all plots in a gridplot layout."

bryevdv commented on sphinx/source/docs/user_guide/export.rst:
```diff
@@ -1,46 +1,44 @@
 .. _userguide_export_png:
 
-PNG Generation
+PNG generation
```
I think this should be "Exporting PNG images"

bryevdv commented on sphinx/source/docs/user_guide/export.rst:
```diff
@@ -68,35 +65,33 @@ saving the image to a file, you can use the lower-level function
 .. _userguide_export_svg:
 
-SVG Generation
+SVG generation
```
I think this should be "Exporting SVG images"

bryevdv commented on sphinx/source/docs/user_guide/export.rst:
```diff
@@ -105,22 +100,20 @@ to ``"svg"``.
     plot.output_backend = "svg"
 
-Exporting an SVG Image
+Exporting an SVG image
```
This whole section is very muddled (has been from the start). I think the "SVG Generation" headline should become "Exporting SVG images" and end with `export_pngs`. Then subsection for "limitations" and maybe finally a section about saving directly from browsers (e.g. with the bookmarlket)

bryevdv commented:
@Konntroll thanks for the PR! I have left some comments. This entire page has always been pretty confused and unorganized since the start, so I tried to outline some changes to the structure. Let me know if you are up for tackling that and if they make sense! 

@tcmetzger would appreciate your input at well!

tcmetzger commented:
Some restructuring would indeed be helpful, I think. @bryevdv the outline you propose makes sense to me. Can you provide more details about the new `export_svg` - is there an up-to-date reference guide entry to use as a basis for updating the user guide?

bryevdv commented:
basic entry here: https://docs.bokeh.org/en/latest/docs/reference/io.html#bokeh.io.export.export_svg

There's not much diff with `export_svgs` except that it can produce one SVG for an entire layout, instead of a separate SVG for every plot in a layout.

Konntroll commented:
Fixed headings and some minor issues and restructured the doc.

I think that with ``export_svg`` / ``export_svgs`` the sub-section on limitation has no purpose, so I dropped it entirely.

I believe that the underlying docs for the two functions have been copy/pasted rather carelessly and so I'm unsure whether the snippets make sense and/or work. Have a look at [``export_svg``](https://docs.bokeh.org/en/latest/docs/reference/io.html#bokeh.io.export.export_svg) and [``export_svgs``](https://docs.bokeh.org/en/latest/docs/reference/io.html#bokeh.io.export.export_svgs) and let me know if these need to be edited. I think, given that the latter should return a list, the ``filename`` parameter should be different somehow.

Let me know what you think.

tcmetzger commented on sphinx/source/docs/user_guide/export.rst:
```diff
@@ -1,46 +1,49 @@
+You can install these dependencies in several ways, including the
+following:
+
+• Using Conda:
```
They might render in a way, because Sphinx just puts the asci character into the html code. However, this is not a good way to generate lists with Sphinx - you should definitely use asterisks as described in the link to the spinx docs.

bryevdv replied:
Right, sorry I was not specific, I meant would they be rendered as actual `<li>` items as an asterisk would be

Konntroll replied:
Okay, noted. Assumed it would be okay to use this character because of how [bullet lists](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#bullet-lists) are described in the reStructuredText Markup Specification.

tcmetzger replied:
Using asterisks also ensures a certain level of consistency throughout the various elements of Bokeh's docs. There are often different ways to do one and the same thing with ReST. When in doubt, I would generally follow the recommendations in the Sphinx documentation.

bryevdv commented:
@Konntroll I think changing the bullets to asterisks is the only thing remaining before this can be merged

bryevdv commented:
Actually, it would be good to make the m"Additional dependencies" section better as well, the current text muddies some things. Here are the requirements to use the export functions:

* Selenium (this is always needed), and
* ONE set of either
  * firefox and geckdriver, OR
  * chrome and chrome driver

All of those can be obtained with conda from conda-forge which is they the current split that only says geckodriver and firefox can be gotten with conda is misleading. I think we should state the basic requirements up above first, then show how both sets can be gotten with conda, and also mention that conda is not the only way.

cc @tcmetzger

Konntroll commented:
Sure, I'll tweak the 'Additional dependencies' section either today or over the weekend.

Konntroll commented:
Implemented the changes. @bryevdv, please have a look at the ``install`` commands in the 'Additional dependencies'. I dropped Firefox from the command you suggested for two reasons:

- The user is likely to have their browser of choice installed.
- Conda / Anaconda don't seem to have a package for Chrome / Chromium.

Let me know if I'm missing something here and I'll make the necessary fixes.

bryevdv commented:
@Konntroll changes look good! There was a missing sphinx substitution for `export_svg`. I'd like to try and merge this tonight so I just pushed the quick change to add it directly
